models/sso.go

```diff
@@ -43,6 +43,47 @@ type SAMLAttributeMapping struct {
 	Keys map[string]SAMLAttribute `json:"keys,omitempty"`
 }
 
+func (m *SAMLAttributeMapping) Equal(o *SAMLAttributeMapping) bool {
+	if m == o {
+		return true
+	}
+
+	if m == nil || o == nil {
```

Comment:
if `m == nil` then calling `m.Equal` would panic right?

Reply:
I don't think so. Go is weird that way... you're actually calling a method on the "variable" and not its contents. 😓 

[Welcome to hell. 🙃](https://go.dev/play/p/EvRWXN4J8C0)